src/esphome/binary_sensor/ttp229_lsf_sensor.cpp

```diff
@@ -0,0 +1,96 @@
+#include "esphome/defines.h"
+#ifdef USE_TTP229_LSF
+
+#include "esphome/binary_sensor/ttp229_lsf_sensor.h"
+#include "esphome/log.h"
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace binary_sensor {
+
+static const char *TAG = "binary_sensor.ttp229";
+
+TTP229Channel::TTP229Channel(const std::string &name, int channel_num) : BinarySensor(name) { channel_ = channel_num; }
+
+void TTP229Channel::process(const uint16_t *data, const uint16_t *last_data) {
+  if ((*data & (1 << this->channel_)) && !(*last_data & (1 << this->channel_))) {
+    this->publish_state(true);
+  }
+  if (!(*data & (1 << this->channel_)) && (*last_data & (1 << this->channel_))) {
+    this->publish_state(false);
+  }
```

Comment:
This could be simplified to 
```cpp
this->publish_state(*data & (1 << this->channel_));
```

logging won't be overrun with messages because binary sensor already filters duplicates.

then you also don't need last_data.